src/OrbitBase/include/OrbitBase/File.h

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_FILE_H_
+#define ORBIT_BASE_FILE_H_
+
+#include <absl/strings/str_format.h>
+#include <fcntl.h>
+
+#include <filesystem>
+#include <string_view>
+
+#include "OrbitBase/Result.h"
+#include "OrbitBase/SafeStrerror.h"
+#include "OrbitBase/UniqueResource.h"
+
+#if defined(__linux)
+#include <unistd.h>
+#elif defined(_WIN32)
+#include <io.h>
+// Windows does not have TEMP_FAILURE_RETRY - define a shortcut
+#define TEMP_FAILURE_RETRY(expression) (expression)
+#endif
+
+namespace orbit_base {
+
+using unique_fd = unique_resource<int, std::function<void(int)>>;
```

Comment:
Since `FdDeleteter` is captureless, have you considered using a function pointer as the deleter?
That would be one indirection less and also takes less space.

`using unique_fd = unique_resource<int, void(*)(int)>;`

Reply:
done.